pymongo/asynchronous/encryption.py

```diff
@@ -194,9 +194,7 @@ async def kms_request(self, kms_context: MongoCryptKmsContext) -> None:
             # Wrap I/O errors in PyMongo exceptions.
             _raise_connection_failure((host, port), error)
 
-    async def collection_info(
-        self, database: AsyncDatabase[Mapping[str, Any]], filter: bytes
-    ) -> Optional[bytes]:
+    async def collection_info(self, database: str, filter: bytes) -> Optional[bytes]:
```

Comment:
Was the signature wrong before?

Reply:
We appear to pass a database name and not an actual database here.

Reply:
And we didn't catch this before because it wasn't in the mypy checks?

Reply:
We don't call `collection_info` anywhere in our own codebase, only in libmongocrypt. That's probably why it didn't get caught.